src/main.cpp

```diff
@@ -3,3 +3,127 @@
  */
 
 #include "Arduino.h"
+#include <SimpleFOC.h>
+
+//  Encoder(int encA, int encB , int cpr, int index)
+//  - encA, encB    - encoder A and B pins
+//  - ppr           - impulses per rotation  (cpr=ppr*4)
+//  - index pin     - (optional input)
+Encoder sensor = Encoder(2, 3, 2048); //will we have hardware or software interrupts?
+
+// interrupt routine initialization
+void doA(){sensor.handleA();}
+void doB(){sensor.handleB();}
+//void doIndex(){encoder.handleIndex();} if we have more than 2 hardware interrupts
+
+// BLDC motor & driver instance
+// For Surpass C2216-880KV motor | 14 pole pairs | 0.108 Resistance | 880KV
+BLDCMotor motor = BLDCMotor(14);
+
+//  BLDCDriver3PWM( int phA, int phB, int phC, int enA, int enB, int enC )
+//  - phA, phB, phC - A,B,C phase pwm pins
+//  - enA, enB, enC - enable pin for each phase (optional)
+BLDCDriver3PWM driver = BLDCDriver3PWM(9, 5, 6, 8);
+
+// commander communication instance
+Commander command = Commander(Serial);
+void onMotor(char* cmd){ command.motor(&motor, cmd); }
+
+void setup() {
+
+  // enable/disable quadrature mode
+  sensor.quadrature = Quadrature::ON;
+
+  // check if you need internal pullups, enable if so
+  // sensor.pullup = Pullup::USE_EXTERN;
+
+  // initialise magnetic sensor hardware
+  sensor.init();
+  // link the motor to the sensor
+  motor.linkSensor(&sensor);
+
+  // enable encoder hardware interrupts
+  sensor.enableInterrupts(doA, doB);
+
+  // driver config
+  driver.voltage_power_supply = 12; // Volts
```

Comment:
Not sure what this is referring to, but the "driver" (the MC PCB) will receive up to 50.4v.

Reply:
👍 Haven't set most parameters just yet. Will update this one and some of the others!